source/MaterialXTest/ShaderValid.cpp

```diff
@@ -449,99 +407,86 @@ TEST_CASE("GLSL MaterialX documents", "[shadervalid]")
 
             std::vector<mx::NodeGraphPtr> nodeGraphs = doc->getNodeGraphs();
             std::vector<mx::OutputPtr> outputList = doc->getOutputs();
-            // TO ADD: Checking for shaders to validate
-            if (!nodeGraphs.empty() || !outputList.empty())
+            std::unordered_set<mx::OutputPtr> outputSet(outputList.begin(), outputList.end());
+            std::vector<mx::MaterialPtr> materials = doc->getMaterials();
+
+            if (!materials.empty() || !nodeGraphs.empty() || !outputList.empty())
             {
                 log << "MTLX Filename: " << filename << std::endl;
+
+                std::unordered_set<mx::OutputPtr> shaderrefOutputs;
+                for (auto material : materials)
+                {
+                    for (auto shaderRef : material->getShaderRefs())
+                    {
+                        if (!shaderRef->hasSourceUri())
+                        {
+                            mx::HwShaderGenerator& hwGenerator = static_cast<mx::HwShaderGenerator&>(*glslShaderGenerator);
```

Comment:
This cast can be moved out from the loop. Can be placed right after shader generator creation.